plugins/modules/ndo_interface_setting.py

```diff
@@ -0,0 +1,837 @@
+            l2_interface_qinq=dict(type="str", choices=list(L2_INTERFACE_QINQ_MAP)),
+            l2_interface_reflective_relay=dict(type="str", choices=["enabled", "disabled"]),
+            stp_bpdu_filter=dict(type="str", choices=["enabled", "disabled"]),
+            stp_bpdu_guard=dict(type="str", choices=["enabled", "disabled"]),
+            llfc_transmit_state=dict(type="str", choices=["enabled", "disabled"]),
+            llfc_receive_state=dict(type="str", choices=["enabled", "disabled"]),
+            mcp=dict(
+                type="dict",
+                options=dict(
+                    admin_state=dict(type="str", choices=["enabled", "disabled"]),
+                    strict_mode=dict(type="str", choices=["on", "off"], aliases=["mcp_mode"]),
+                    initial_delay_time=dict(type="int"),
+                    transmission_frequency_sec=dict(type="int"),
+                    transmission_frequency_msec=dict(type="int"),
+                    grace_period_sec=dict(type="int"),
+                    grace_period_msec=dict(type="int"),
+                ),
+            ),
+            pfc_admin_state=dict(type="str", choices=["on", "off", "auto"]),
+            access_macsec_policy=dict(type="str"),
+            state=dict(type="str", choices=["absent", "query", "present"], default="query"),
+        )
+    )
+
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        supports_check_mode=True,
+        required_if=[
+            ["state", "present", ["interface_policy_group", "interface_policy_group_uuid", "interface_type"], True],
+            ["state", "absent", ["interface_policy_group", "interface_policy_group_uuid"], True],
+        ],
+    )
+
+    mso = MSOModule(module)
+
+    template = module.params.get("template")
+    interface_policy_group = module.params.get("interface_policy_group")
+    interface_policy_group_uuid = module.params.get("interface_policy_group_uuid")
+    description = module.params.get("description")
+    interface_type = module.params.get("interface_type")
+    if interface_type == "port_channel":
+        interface_type = "portchannel"
+    speed = module.params.get("speed")
+    auto_negotiation = module.params.get("auto_negotiation")
+    if auto_negotiation == "on_enforce":
+        auto_negotiation = "on-enforce"
+    vlan_scope = module.params.get("vlan_scope")
+    if vlan_scope == "port_local":
+        vlan_scope = "portlocal"
+    cdp_admin_state = module.params.get("cdp_admin_state")
+    lldp = module.params.get("lldp")
+    domains = module.params.get("domains")
+    port_channel_mode = PORT_CHANNEL_MODE_MAP.get(module.params.get("port_channel_mode"))
+    min_links = module.params.get("min_links")
+    max_links = module.params.get("max_links")
+    if module.params.get("control") is not None:
+        control = sorted(CONTROL_MAP.get(v) for v in module.params.get("control"))
+    else:
+        control = None
```

Comment:
why not keep it consistent like with your other assignment if statements?
```suggestion
    control = module.params.get("control")
    if control: # sort only needed when list is not empty and not none
        control = sorted(CONTROL_MAP.get(v) for v in module.params.get("control"))
```

Reply:
changed it to keep it consistent